cmake/external/onnx_minimal.cmake

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+#
+# Setup onnx and onnx_protobuf for a build with onnxruntime_MINIMAL_BUILD enabled.
+# We exclude everything but the essentials from the onnx library.
+#
+
+#TODO: if protobuf is a shared lib and onnxruntime_USE_FULL_PROTOBUF is ON, then onnx_proto should be built as a shared lib instead of a static lib. Otherwise any code outside onnxruntime.dll can't use onnx protobuf definitions if they share the protobuf.dll with onnxruntime. For example, if protobuf is a shared lib and onnx_proto is a static lib then onnxruntime_perf_test won't work.
+
+set(ONNX_SOURCE_ROOT ${PROJECT_SOURCE_DIR}/external/onnx)
+
+add_library(onnx_proto ${ONNX_SOURCE_ROOT}/onnx/onnx-ml.proto ${ONNX_SOURCE_ROOT}/onnx/onnx-operators-ml.proto)
+
+target_include_directories(onnx_proto PUBLIC $<TARGET_PROPERTY:protobuf::libprotobuf,INTERFACE_INCLUDE_DIRECTORIES> "${CMAKE_CURRENT_BINARY_DIR}")
+target_compile_definitions(onnx_proto PUBLIC $<TARGET_PROPERTY:protobuf::libprotobuf,INTERFACE_COMPILE_DEFINITIONS>)
+
+set(_src_prefix "onnx/")
+onnxruntime_protobuf_generate(NO_SRC_INCLUDES GEN_SRC_PREFIX ${_src_prefix} IMPORT_DIRS ${ONNX_SOURCE_ROOT} TARGET onnx_proto)
+
+if (WIN32)
+  target_compile_options(onnx_proto PRIVATE "/wd4146" "/wd4125" "/wd4456" "/wd4267" "/wd4309")
+else()
+  if(HAS_UNUSED_VARIABLE)
+    target_compile_options(onnx_proto PRIVATE "-Wno-unused-variable")
+  endif()
+
+  if(HAS_UNUSED_BUT_SET_VARIABLE)
+    target_compile_options(onnx_proto PRIVATE "-Wno-unused-but-set-variable")
+  endif()   
+endif()
+
+set_target_properties(onnx_proto PROPERTIES FOLDER "External/ONNX")
+
+# Cpp Tests were added and they require googletest
+# since we have our own copy, try using that
+if(NOT onnxruntime_MINIMAL_BUILD)
+  file(GLOB_RECURSE onnx_src CONFIGURE_DEPENDS
+      "${ONNX_SOURCE_ROOT}/onnx/*.h"
+      "${ONNX_SOURCE_ROOT}/onnx/*.cc"
+  )
+  file(GLOB_RECURSE onnx_exclude_src CONFIGURE_DEPENDS
+      "${ONNX_SOURCE_ROOT}/onnx/py_utils.h"
+      "${ONNX_SOURCE_ROOT}/onnx/proto_utils.h"
+      "${ONNX_SOURCE_ROOT}/onnx/backend/test/cpp/*"
+      "${ONNX_SOURCE_ROOT}/onnx/test/*"
+      "${ONNX_SOURCE_ROOT}/onnx/cpp2py_export.cc"
+  )
+  list(REMOVE_ITEM onnx_src ${onnx_exclude_src})  
```

Comment:
Remove this and add failure here something like,
if(NOT onnxruntime_MINIMAL_BUILD)
  message(FATAL_ERROR "onnxruntime_MINIMAL_BUILD required to build onnx_minimal")

Reply:
we don't include this file unless it's a minimal build though